src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/CngSymmetricAlgorithmCore.cs

```diff
@@ -235,6 +274,27 @@ private void ValidateFeedbackSize(CipherMode mode, int feedbackSizeInBits)
             }
         }
 
+        private byte[] CopyAndValidateKey(byte[] rgbKey)
+        {
+            byte[] key = rgbKey.CloneByteArray();
+
+            long keySize = key.Length * (long)BitsPerByte;
+            if (keySize > int.MaxValue || !((int)keySize).IsLegalSize(_outer.LegalKeySizes))
+            {
+                throw new ArgumentException(SR.Cryptography_InvalidKeySize, nameof(rgbKey));
+            }
```

Comment:
Also, once we're moving things around, let's move the throw above the clone/ToArray.  No point in duplicating something just to throw it away when we can precondition on something else.